impl/src/main/java/org/jboss/weld/util/Beans.java

```diff
@@ -583,6 +583,13 @@ public static boolean hasBuiltinScope(Bean<?> bean) {
                 || ConversationScoped.class.equals(bean.getScope()) || Dependent.class.equals(bean.getScope());
     }
 
+    public static boolean hasBuiltinScope(AnnotatedType<?> annotatedType) {
+        return annotatedType.getAnnotations().contains(RequestScoped.Literal.INSTANCE) || annotatedType.getAnnotations()
```

Comment:
How about storing `annotatedType.getAnnotations()` into a variable and write the boolean expression using that? It would make the boolean expression much shorter.